feat: list.item support styles

wanpan11 opened the pull request:
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close https://github.com/ant-design/ant-design/issues/48372

### 💡 Background and solution

List.Item 增加 styles 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      feat: list.item support styles     |
| 🇨🇳 Chinese |      List.Item 增加 styles      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

wanpan11 committed:
feat: list.item support styles

wanpan11 committed:
feat: list.item support styles

wanpan11 committed:
feat: update

wanpan11 committed:
feat: update

wanpan11 committed:
feat: update

wanpan11 committed:
feat: update

li-jia-nan commented on components/list/Item.tsx:
```diff
@@ -116,11 +121,17 @@ const InternalItem = React.forwardRef<HTMLDivElement, ListItemProps>((props, ref
+        : [
+            children,
+            actionsContent,
+            <span key="extra" style={styles?.extra}>
```
这样的话是不是 dom 结构发生变化了？

直接给 `cloneElement` 加 `style` 属性可以吗？

```ts
cloneElement(extra, { key: 'extra', style: styles?.extra })
```

li-jia-nan replied:
那最好判断一下，如果 `typeof extra === "string"` 的时候再加 `span`，这样是不是对 dom 结构破坏小一点？

wanpan11 replied:
阔以

li-jia-nan replied:
另外如果这个节点是 string 的话，那么就不是一个有效的 ReactElement，这时候原来的 key 也是不生效的

wanpan11 committed:
feat: optimized code

wanpan11 committed:
feat: optimized code

Wxh16144 commented on components/list/Item.tsx:
```diff
@@ -116,11 +121,18 @@ const InternalItem = React.forwardRef<HTMLDivElement, ListItemProps>((props, ref
+        : [
+            children,
+            actionsContent,
+            cloneElement(typeof extra === 'string' ? <span>{extra}</span> : extra, {
```
`typeof extra === 'string' ? <span>{extra}</span> : extra` 这个逻辑看起来很奇怪, 像是加了一个 feature 套了一个 div，并且也没用加 key.

wanpan11 replied:
目前 extra 有可能会是 string 所以套了个 span ，后面的props加了 key 呀，

MadCcc replied:
这里可以不处理，语义化的场景必须和 class 绑定，强行加 span 不是一个好的方法，因为用户完全可以自己写 span

wanpan11 replied:
> 这里可以不处理，语义化的场景必须和 class 绑定，强行加 span 不是一个好的方法，因为用户完全可以自己写 span

好的 这里好像有个问题，文档上说 horizontal 下 extra 会在最有侧
![image](https://github.com/ant-design/ant-design/assets/38714194/b4931a6c-7f44-4f14-b18d-a794a6818fee)

但是 horizontal  下如果没有传 actions 的话 extra 是会紧跟在后面的（好像就是因为 extra 没有外壳导致的）
![image](https://github.com/ant-design/ant-design/assets/38714194/9a1fa730-0b6c-41b2-93ff-fad5bf042323)

crazyair committed:
fix: fix Checkbox.Group type (#49073)

* fix: fix Checkbox.Group type

* fix: fix Checkbox.Group type

* fix: fix Checkbox.Group type

* fix: remove type

* fix: any

zombieJ committed:
docs: 5.17.4 changelog (#49076)

* docs: 5.17.4 changelog

* docs: more

li-jia-nan committed:
site: update site scrollbar style (#49088)

wanpan11 committed:
Merge branch 'feature' into feat-list-item

liangchaofei committed:
docs: update antd pro i18n link (#49098)

Co-authored-by: liangchaofei <liangchaofei@liangchaofeideMacBook-Pro.local>

crazyair committed:
docs: add getValueProps demo (#49072)

* docs: add getValueProps demo

* feat: normalize

* feat: normalize

* feat: name

* feat: name

* feat: doc

* feat: date

* feat: tz

* feat: tz

MadCcc commented:
语义化的 classNames 也可以支持下，和 styles 一致。另外可以在 ConfigProvider 上透出

selicens committed:
docs(FAQ): Remove expired links (#49082)

* docs(FAQ): Remove expired links

* chore: adjusting update

* chore: adjusting update

JeevaRamanathan committed:
docs: Updated typo of selected (#49104)

Signed-off-by: JeevaRamanathan <64531160+JeevaRamanathan@users.noreply.github.com>

github-actions[bot] committed:
chore: upgrade deps (#49103)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

afc163 committed:
fix: Tour onClose wrong current argument (#49124)

* fix: Tour onClose wrong current argument

* test: add test case

* Update components/tour/panelRender.tsx

Signed-off-by: afc163 <afc163@gmail.com>

---------

Signed-off-by: afc163 <afc163@gmail.com>

github-actions[bot] committed:
chore: auto merge branches (#49137)

chore: feature merge master

zombieJ committed:
test: Update test of demo (#49140)

* chore: rollback ci

* test: one day

zombieJ committed:
refactor: Watermark add overflow hidden style to avoid visible show (#49130)

* refactor: add overflow style

* test: add test case

crazyair committed:
feat: Drawer support drawerRender (#49125)

* feat: Drawer support drawerRender

* feat: add test

kelisiWu123 committed:
docs: Modify inappropriate words (#49143)

Co-authored-by: 吴必炫 <wubixuan@rlink.com>

mySkey committed:
fix: Anchor - The replace attribute causes no scrolling animation (#49136)

Co-authored-by: dengxia <dengxia@tanqingsk.com>

dependabot[bot] committed:
chore(deps-dev): bump @types/react-highlight-words from 0.16.7 to 0.20.0 (#49142)

Bumps [@types/react-highlight-words](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react-highlight-words) from 0.16.7 to 0.20.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react-highlight-words)

---
updated-dependencies:
- dependency-name: "@types/react-highlight-words"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Wxh16144 committed:
feat: Rate support keyboard prop (#49132)

* feat: rate add support for disable keyboard events

* chore: bump rc-rate@2.13.0

* chore: update

coding-ice committed:
type: remove unnecessary types (#49146)

LemonTency committed:
docs: Update AutoComplete index.zh-CN.md (#49139)

* Update index.zh-CN.md

Signed-off-by: LemonTency <120943952@qq.com>

* Update index.zh-CN.md

Signed-off-by: LemonTency <120943952@qq.com>

---------

Signed-off-by: LemonTency <120943952@qq.com>

crazyair committed:
Form.Item support layout (#49119)

* feat: FormItem support layout

* feat: doc

* feat: test

* feat: test

* feat: test

* feat: test

* feat: itemPrefixCls

afc163 committed:
Merge pull request #49145 from ant-design/master

chore: feature merge master

wanpan11 committed:
Merge branch 'feat-list-item' of https://github.com/wanpan11/ant-design into feat-list-item
